### PR TITLE
client: set sensible defaults for Browse

### PR DESCRIPTION
**Pull request by magiconair**
This patch sets sensible defaults for the `Browse` method of the client.

See #506

### CODE REVIEW

**Comment by kung-foo** [comment]
These "sane defaults" are fine, however does this mean we should really do this for _all_ *Requests.

**Comment by kung-foo** [comment]
LGTM once merge conflict resolved.

**Review by kung-foo — APPROVED** [review]

**Comment by magiconair** [comment]
Well, we did this for `Read` and we have `client.Browse` and `node.Browse` which is confusing. I think other libraries solve this by providing defaults in arguments or helper methods. I'm OK with solving this in another way if you have a suggestion.

**Comment by kung-foo** [comment]
I guess the issue is that "the struct is the api". Since it is unlikely that we want to make everything private (i.e. _only_ make a `BrowseRequest` instance via builders/helpers) then we need strong, public validators and/or implicit defaults.

**Pull request merged**
